### PR TITLE
Feature/async git helper

### PR DESCRIPTION
**Pull request by bezoerb**
Solves #3553

### CODE REVIEW

File: gulpfile.js/build.js [hunk 1]
```diff
@@ -529,18 +529,22 @@ function collectStatics(done) {
...
-    'by': git.user,
-    'environment': config.environment,
-    'commit': {
-      'sha': git.version,
-      'message': git.message,
-    },
-  };
-
-  fs.writeFile(project.paths.BUILD_INFO, yaml.safeDump(buildInfo), done);
+  Promise.all([git.user(), git.version(), git.message()])
```
Comment (patrickkettner) [inline gulpfile.js/build.js:532]: we are using async/await elsewhere in the file - any reason you aren't here?

File: gulpfile.js/deploy.js [hunk 2]
```diff
@@ -35,7 +35,7 @@ const TIMESTAMP = Date.now();
 // We tag docker images by the current git commit SHA + TIMESTAMP,
 // this makes it easy to identify and reproduce builds.
 // Pass a specific tag via commandline using `gulp updateStart --tag ABCDE...`
-const TAG = argv.tag || `${require('@lib/utils/git').version}-${TIMESTAMP}`;
+const TAG = argv.tag || `${require('@lib/utils/git').version()}-${TIMESTAMP}`;
```
Comment (patrickkettner) [inline gulpfile.js/deploy.js:38]: this would resolve as a `string || Promise`, no?

File: platform/lib/utils/sh.js [hunk 3]
```diff
@@ -44,7 +44,9 @@ function sh(commandLine, ...options) {
   const command = fragments[0];
   const args = fragments.splice(1);
 
-  console.log(`$ ${command} ${args.join(' ')}`);
+  if (!opts.quiet) {
+    console.log(`$ ${command} ${args.join(' ')}`);
```
Comment (patrickkettner) [inline platform/lib/utils/sh.js:48]: would it make sense to use the [util logger](https://github.com/ampproject/amp.dev/blob/future/platform/lib/utils/log.js)?

File: gulpfile.js/build.js (hunk 1 above)
Comment (bezoerb) [reply gulpfile.js/build.js:532]: Didn't want to change the the function declaration ;)
I'll give it a try

File: gulpfile.js/deploy.js (hunk 2 above)
Comment (bezoerb) [reply gulpfile.js/deploy.js:38]: good catch :)

File: platform/lib/utils/sh.js (hunk 3 above)
Comment (bezoerb) [reply platform/lib/utils/sh.js:48]: The logger util is used to provide scoped logs. I would keep the `console.log` statements as we don't have a specific scope here

File: gulpfile.js/build.js [hunk 4]
```diff
@@ -528,23 +535,23 @@ function collectStatics(done) {
...
-      fs.writeFile(project.paths.BUILD_INFO, yaml.safeDump(buildInfo), done);
-    })
-    .catch((error) => done(error));
+async function persistBuildInfo() {
+  const version = await git.version();
+  const user = await git.user();
+  const message = await git.message();
+
+  const buildInfo = {
+    'number': travis.build.number || null,
```
Comment (patrickkettner) [inline gulpfile.js/build.js:544]: when would `travis.build.number` not be defined? If it isn't, are we sure `travis.build` exists?

Comment (patrickkettner) [reply gulpfile.js/build.js:544]: ping @bezoerb

Comment (matthiasrohmer) [reply gulpfile.js/build.js:544]: Yep, `travis.build` (and even `.number`) will always exist though may be undefined:

https://github.com/ampproject/amp.dev/blob/4f010c325688cf2c1037f2e56d1b0d3ed159fd4e/platform/lib/utils/travis.js#L51-L54

**Review by patrickkettner — APPROVED** [review]

**Pull request merged**